common/util/autoware_utils/include/autoware_utils/trajectory/convert.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE_UTILS__TRAJECTORY__CONVERT_HPP_
+#define AUTOWARE_UTILS__TRAJECTORY__CONVERT_HPP_
+
+#include "autoware_utils/geometry/geometry.hpp"
+#include "autoware_utils/geometry/pose_deviation.hpp"
+
+#include "autoware_auto_planning_msgs/msg/trajectory.hpp"
+#include "autoware_auto_planning_msgs/msg/trajectory_point.hpp"
+
+#include <boost/optional.hpp>
+
+#include <algorithm>
+#include <vector>
+
+namespace autoware_utils
+{
+using autoware_auto_planning_msgs::msg::Trajectory;
+using TrajectoryPointArray = std::vector<autoware_auto_planning_msgs::msg::TrajectoryPoint>;
+
+// convertToTrajectoryByClipping() just clips TrajectoryPointArray up to the capacity of Trajectory.
+// Therefore, the error handling out of this function is necessary if the size of
+// TrajectoryPointArray greater than the capacity
```

Comment:
Thank you for adding this comment, but we can't judge whether these functions should be removed or not by this when we'll create Autoware Core.
Please add more definite comments.　:pray: 

At least, you should add some `TODO` comments.

Reply:
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/512/commits/e193d49eeb49c8e1deb3a06745cea2e3734ad7e0

Now is it merge ready? @kenji-miyake

Reply:
TODO style is wrong. (but there is no error :thinking: )
https://ttsuki.github.io/styleguide/cppguide.ja.html#TODO_Comments